lib/ansible/modules/cloud/docker/docker_plugin.py

```diff
@@ -0,0 +1,297 @@
+            else:
+                existing_options_list = self.existing_plugin.settings['Env']
+                existing_options = parse_options(existing_options_list)
+
+                for key, value in iteritems(self.parameters.plugin_options):
+                    if ((not existing_options.get(key) and value) or
+                            not value or
+                            value != existing_options[key]):
+                        differences.append('plugin_settings.%s' % key)
+
+        return differences
+
+    def install_plugin(self):
+        if not self.existing_plugin:
+            if not self.check_mode:
+                try:
+                    self.existing_plugin = self.dclient.plugins.install(self.parameters.plugin_name, self.parameters.plugin_alias)
+                except APIError as e:
+                    self.client.fail(text_type(e))
+
+            self.results['actions'].append("Installed plugin %s" % self.get_plugin_name())
+            self.results['changed'] = True
+
+    def remove_plugin(self):
+        if self.existing_plugin:
+            if not self.check_mode:
+                try:
+                    self.existing_plugin.remove()
+                except APIError as e:
+                    self.client.fail(text_type(e))
+
+            self.results['actions'].append("Removed plugin %s" % self.get_plugin_name())
+            self.results['changed'] = True
+
+    def update_plugin(self):
+        if not self.check_mode:
+            try:
+                if self.existing_plugin.enabled:
+                    self.existing_plugin.disable()
+                self.existing_plugin.configure(prepare_options(self.parameters.plugin_options))
+                self.existing_plugin.enable(1)
+            except APIError as e:
+                self.client.fail(text_type(e))
+
+        self.results['actions'].append("Updated plugin %s settings" % self.get_plugin_name())
+        self.results['changed'] = True
+
+    def present(self):
+        differences = []
+        if self.existing_plugin:
+            differences = self.has_different_config()
+
+        if self.existing_plugin:
+            self.update_plugin()
+        else:
+            self.install_plugin()
+
+        if self.diff or self.check_mode or self.parameters.debug:
+            self.results['diff'] = differences
+
+        if not self.check_mode and not self.parameters.debug:
+            self.results.pop('actions')
+
+    def absent(self):
+        self.remove_plugin()
+
+    def enable(self):
+        if self.existing_plugin:
+            if not self.check_mode:
+                try:
+                    self.existing_plugin.enable(1)
+                except APIError as e:
+                    self.client.fail(text_type(e))
+
+            self.results['actions'].append("Enabled plugin %s" % self.get_plugin_name())
+            self.results['changed'] = True
+        else:
+            self.fail("Cannot enable plugin: Plugin not exist")
```

Comment:
`enabled` should mean the same as `present` and make sure the plugin is enabled, i.e. if it is not installed, install it. (Same for `disabled`.) `present` means "make sure it is there, I don't care if enabled or not".

Reply:
If for some reason you want to be able to enable without installing, you could add a flag which for `enable`/`disable` fails instead of installing (if the plugin doesn't exist). But the default behavior should be "install if not there, then make sure you have the correct state".